deepchem/models/layers.py

```diff
@@ -471,18 +471,87 @@ def call(self, inputs):
 
 
 def _cosine_dist(x, y):
-  """Computes the inner product (cosine distance) between two tensors.
+  """Computes the inner product (cosine similarity) between two tensors.
+  
+  This assumes that the two input tensors contain rows of vectors where 
+  each column represents a different feature. The output tensor will have
+  elements that represent the inner product between pairs of normalized vectors
+  in the rows of `x` and `y`. The two tensors need to have the same number of columns,
+  because one cannot take the dot product between vectors of different lengths.
+  For example, in sentence similarity and sentence classification tasks,
+  the number of columns is the embedding size. In these tasks, the rows of the
+  input tensors would be different test vectors or sentences. The input tensors
+  themselves could be different batches. Using vectors or tensors of all 0s
+  should be avoided.
+
+  Method
+  ------
+  The vectors in the input tensors are first l2-normalized such that each vector
+  has length or magnitude of 1. The inner product (dot product) is then taken 
+  between corresponding pairs of row vectors in the input tensors and returned.
+
+  Examples
+  --------
+  The cosine similarity between two equivalent vectors will be 1. The cosine
+  similarity between two equivalent tensors (input tensors where the elements are
+  the same), will be a tensor of 1s. In this scenario, if the input tensors
+  `x` and `y` were each of shape `(n,p)`, where each element in `x` and `y`
+  were the same, then the output tensor would be a tensor of shape `(n,n)` 
+  with 1 in every entry.
+  
+  >>> import tensorflow as tf
+  >>> import deepchem.models.layers as layers
+  >>> x = tf.ones((6, 4), dtype=tf.dtypes.float32, name=None)
+  >>> y_same = tf.ones((6, 4), dtype=tf.dtypes.float32, name=None)
+  >>> # x and y are the same tensor (equivalent at every element)
```

Comment:
Instead of having theses as comments, you can break out a little paragraph of text. That might make it easier for readers to follow along

Reply:
Sounds good, this should be changed now : )